src/interpreter-errors.ts

```diff
@@ -117,39 +133,67 @@ export class UnassignedVariable extends RuntimeSourceError {
   }
 
   public explain() {
-    return `Name ${this.name} not yet assigned`
+    return `Name ${this.name} not yet assigned.`
   }
 
   public elaborate() {
-    return 'TODO'
+    return `Please assign a value to ${this.name} first.\n\n\t${this.name} = 42;`
   }
 }
 
 export class InvalidNumberOfArguments extends RuntimeSourceError {
+  private calleeStr: string
+
   constructor(node: es.Node, private expected: number, private got: number) {
     super(node)
+    this.calleeStr = ((node as es.CallExpression).callee as es.Identifier).name
   }
 
   public explain() {
-    return `Expected ${this.expected} arguments, but got ${this.got}`
+    return `Expected ${this.expected} arguments, but got ${this.got}.`
   }
 
   public elaborate() {
-    return 'TODO'
+    const calleeStr = this.calleeStr
+    const pluralS = this.expected === 1 ? '' : 's'
+
+    return `Try calling function ${calleeStr} again in the same way, but with ${
+      this.expected
+    } argument${pluralS} instead. Remember that arguments are separated by a ',' (comma).`
   }
 }
 
 export class VariableRedeclaration extends RuntimeSourceError {
-  constructor(node: es.Node, private name: string) {
+  constructor(private node: es.Node, private name: string) {
     super(node)
   }
 
   public explain() {
-    return `Redeclaring name ${this.name}`
+    return `Redeclaring name ${this.name}.`
   }
 
   public elaborate() {
-    return 'TODO'
+    const elabStr = `Since ${
+      this.name
+    } has already been declared, you can assign a value to it without re-declaring.`
```

Comment:
To take into account trying to redeclare constants in Source? It might be confusing to suggest reassigning a value to a constant.